### PR TITLE
Fix/zios 11316 cannot open a non connected user profile with deep link

### PR DESCRIPTION
**Pull request by billypchan**
## What's new in this PR?

Create a new enum value `.connectToUser(id: UUID, searchTask: SearchTask)` when the user opens a deep link with a user id of an unconnected user.

### CODE REVIEW

File: Source/UserSession/Search/SearchTask.swift [hunk 1]
```diff
@@ -94,6 +94,17 @@ public class SearchTask {
     }
 }
 
+extension SearchTask: Equatable {
```
Comment (alexaubry) [inline Source/UserSession/Search/SearchTask.swift]: This seems to be unrelated to the change. Do we need to make it equatable?

Comment (billypchan) [reply Source/UserSession/Search/SearchTask.swift]: Since I added 'case connectToUser(id: UUID, searchTask: SearchTask)' to `URLAction` and `URLActio` is `Equatable`

Comment (alexaubry) [reply Source/UserSession/Search/SearchTask.swift]: Ah, makes sense now!

**Comment by billypchan** [comment]
close this PR, this can be done with a simpler approach.
